axlearn/common/checkpointer_orbax_emergency_replicator.py

```diff
@@ -0,0 +1,548 @@
+            non_tensor_async_timeout_secs: Timeout for async barrier in seconds when saving
+                non-tensor states.
+            async_timeout_secs: Timeout for async barrier in seconds when saving tensors.
+            replica_axis_index: The index of the "data" axis.
+        """
+
+        keep_last_n: int = 1
+        keep_every_n_steps: Optional[int] = None
+        local_keep_last_n: int = 1
+        local_save_policy: InstantiableConfig[CheckpointPolicy] = config_for_function(
+            every_n_steps_policy
+        ).set(n=10)
+        local_dir: str = "/checkpoint"
+        trainer_dir: Required[str] = REQUIRED
+        non_tensor_async_timeout_secs: int = 300
+        async_timeout_secs: int = 3600
+        replica_axis_index: Required[int] = REQUIRED
+
+    # @classmethod
+    # def checkpoint_paths(cls, base_dir: str) -> List[str]:
+    #    """See `BaseCheckpointer.checkpointer_paths`.
+    #
+    #    Only persistent checkpoint paths are returned. There's no guarantee that the paths returned
+    #    have committed TF savables. Use `checkpoint_steps` to get steps with both tensors and
+    #    committed TF savables.
+    #    """
+    #    logging.log_first_n(
+    #        logging.WARNING,
+    #        msg="checkpoint_paths is deprecated. Use checkpoint_steps instead.",
+    #        n=1,
+    #    )
+    #    tensors_dir = os.path.join(base_dir, cls._TENSORS_PREFIX)
+    #    return [str(path) for path in ocp.utils.checkpoint_steps_paths(tensors_dir)]
+    #
+    # @classmethod
+    # def checkpoint_steps(cls, base_dir) -> list[int]:
+    #    """See `BaseCheckpointer.checkpointer_steps`.
+    #
+    #    Only persistent checkpoint steps are returned.
+    #    """
+    #    return list(
+    #        set(
+    #            ocp.utils.checkpoint_steps(os.path.join(base_dir, cls._TENSORS_PREFIX))
+    #        ).intersection(
+    #            set(Checkpointer.checkpoint_steps(os.path.join(base_dir, cls._NON_TENSORS_PREFIX)))
+    #        )
+    #    )
+
+    def __init__(self, cfg: Config, *, parent: Optional[Module]):
+        super().__init__(cfg, parent=parent)
+        cfg: OrbaxEmergencyReplicatorCheckpointer.Config = self.config
+        self._name_format = ocp.step.standard_name_format(
+            step_prefix=None,
+            step_format_fixed_length=None,
+        )
+        # if jax.process_index() == 0:
+        #     fs.makedirs(os.path.join(cfg.dir, self._NON_TENSORS_PREFIX))
+        #     fs.makedirs(os.path.join(cfg.dir, self._TENSORS_PREFIX))
+        self._local_dir = cfg.local_dir
+        fs.makedirs(self._local_dir)
+        # Orbax replicator ckpt requires this function to be called prior to checkpointer
+        # operations. This function also serves as a barrier.
+        ocp.multihost.initialize_runtime_to_distributed_ids()
+        ocp.multihost.initialize_distributed_to_device_ids()
+
+        num_slices = int(os.environ["MEGASCALE_NUM_SLICES"])
+        # De-hardcode
```

Comment:
I'd say it's simply always true here, so remove if

Reply:
relic of maxtext implementation where this is part of emergency checkpointer and so needs that conditional